sandbox_dns_unix.go

```diff
@@ -184,6 +187,13 @@ func (sb *sandbox) setupDNS() error {
 			return err
 		}
 	} else {
+		// If embedded DNS server is enabled later for this contianer ipv4 loopback nameservers if
+		// present should be used. retain it before filtering out resolv.conf contents.
```

Comment:
I dont quite understand this scenario. Could you please explain the basics here ?
It seems like the `sb.extDNS` is cached in `setupDNS` now and not `rebuildDNS`. But beyond that, it is very hard to understand this logic.